tool/tbot/anonymous_telemetry.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/gen/proto/go/prehog/v1alpha"
+	"github.com/gravitational/teleport/gen/proto/go/prehog/v1alpha/v1alphaconnect"
+	"github.com/gravitational/teleport/lib/tbot/config"
+)
+
+const (
+	anonymousTelemetryEnabledEnv = "TELEPORT_ANONYMOUS_TELEMETRY"
+	anonymousTelemetryAddressEnv = "TELEPORT_ANONYMOUS_TELEMETRY_ADDRESS"
+
+	helperEnv        = "_TBOT_TELEMETRY_HELPER"
+	helperVersionEnv = "_TBOT_TELEMETRY_HELPER_VERSION"
+
+	telemetryDocs = "example.com/placeholder"
+)
+
+type envGetter func(key string) string
+
+func telemetryEnabled(envGetter envGetter) bool {
+	if val, err := strconv.ParseBool(
+		envGetter(anonymousTelemetryEnabledEnv),
+	); err == nil {
+		return val
+	}
+	return false
+}
+
+func telemetryClient(envGetter envGetter) v1alphaconnect.TbotReportingServiceClient {
+	// staging: https://reporting-staging.teleportinfra.dev
+	endpoint := "https://reporting.teleportinfra.sh"
```

Comment:
To give us as much flexibility as possible, I feel like we should use different subdomains for the different binaries we ship. WDYT?

Reply:
d11adb1849

I'll raise a PR elsewhere shortly to create these cnames